### PR TITLE
set upper bound on google-cloud-profiler

### PR DESCRIPTION
**Pull request by AnandInguva**
`google-cloud-profiler==4.0.0` is not working with `protobuf==3.19.4`.  Downgrade `google-cloud-profiler` since [tensorflow<=2.10.0](https://github.com/tensorflow/tensorflow/blob/359c3cdfc5fabac82b3c70b3b6de2b0a8c16874f/tensorflow/tools/pip_package/setup.py#L96) requires `protobuf<3.20` so we can't update `protobuf`

Tracking dependency issue on Protobuf, tensorflow and cloud profiler: https://github.com/apache/beam/issues/23355

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by AnandInguva** [comment]
Run Python 3.7 PostCommit

**Comment by AnandInguva** [comment]
Run Python 3.8 PostCommit

**Comment by AnandInguva** [comment]
Run Python 3.9 PostCommit

**Comment by AnandInguva** [comment]
R: @tvalentyn @Abacn

**Review by Abacn — APPROVED** [review]
LGTM

may want to re-run python precommit test before merge, though the error "pytest internal error" most likely irrelevant.

**Comment by AnandInguva** [comment]
Run Python PreCommit

**Comment by AnandInguva** [comment]
Run Python PreCommit

**Comment by Abacn** [comment]
Fine, failure different each time, are unrelated flakes

**Comment by AnandInguva** [comment]
> Fine, failure different each time, are unrelated flakes

Yep.

**Comment by AnandInguva** [comment]
Run Python PreCommit

File: sdks/python/container/base_image_requirements_manual.txt [hunk 1]
```diff
@@ -31,7 +32,7 @@ cython<1
 # some versions of libraries that lauch Beam pipelines, like tensorflow-transform.
 # Leaving 'future' in our containers for now prevent breaking tft users.
 future
-google-cloud-profiler
+google-cloud-profiler<4.0.0
```
Comment (tvalentyn) [inline sdks/python/container/base_image_requirements_manual.txt:35]: Add a comment when can this be lifted.

**Review by tvalentyn — APPROVED** [review]

**Comment by lukecwik** [comment]
I think we should revert this as the tests didn't pass before merging and it seems like there are similar failures causing [beam_PreCommit_Python_Cron to be permared](https://ci-beam.apache.org/job/beam_PreCommit_Python_Cron/6174/#showFailuresLink)

**Comment by lukecwik** [comment]
Nvm, this is due to the objsize 0.6.0 release.

**Pull request merged**